src/ui/components/pages/swap/swap.tsx

```diff
@@ -97,223 +97,219 @@ export function Swap() {
-                minReceivedCoins: minReceivedCoins.toFixed(),
-                slippageTolerance,
-                status: 'lessThanExpected',
-                toAssetId,
-                toCoins: toCoins.toFixed(),
-                vendor,
-              });
-            }
-
-            match = msg.match(
-              /amount to receive is lower than expected one (\d+)/i
-            );
-
-            if (match) {
-              capture = false;
-
-              const expectedAmountCoins = new BigNumber(match[1]);
-
-              background.sendEvent('swapAssets', {
-                expectedAmountCoins: expectedAmountCoins.toFixed(),
-                fromAssetId,
-                fromCoins: fromCoins.toFixed(),
-                minReceivedCoins: minReceivedCoins.toFixed(),
-                slippageTolerance,
-                status: 'lessThanExpected',
-                toAssetId,
-                toCoins: toCoins.toFixed(),
-                vendor,
-              });
-            }
-
-            if (capture) {
-              captureException(new Error(msg));
-            }
-            return;
+            capture = false;
+
+            const expectedAmountCoins = new BigNumber(match[1]);
+
+            background.sendEvent('swapAssets', {
+              expectedAmountCoins: expectedAmountCoins.toFixed(),
+              fromAssetId,
+              fromCoins: fromCoins.toFixed(),
+              minReceivedCoins: minReceivedCoins.toFixed(),
+              slippageTolerance,
+              status: 'lessThanExpected',
+              toAssetId,
+              toCoins: toCoins.toFixed(),
+              vendor,
+            });
           }
 
-          if (errMessage === 'Request is rejected on ledger') {
-            setIsSwapInProgress(false);
-            return;
+          if (capture) {
+            captureException(new Error(msg));
           }
+          return;
+        }
 
-          setSwapErrorMessage(errMessage || t('swap.failMessage'));
+        if (errMessage === 'Request is rejected on ledger') {
           setIsSwapInProgress(false);
+          return;
+        }
 
-          if (capture) {
-            captureException(new Error(errMessage));
-          }
+        setSwapErrorMessage(errMessage || t('swap.failMessage'));
+        setIsSwapInProgress(false);
+
+        if (capture) {
+          captureException(new Error(errMessage));
         }
-      }}
-    >
-      {({ onPrepare, pending }) => (
-        <SwapForm
-          initialFromAssetId={initialFromAssetId}
-          initialToAssetId={initialToAssetId}
-          isSwapInProgress={pending || isSwapInProgress}
-          swapErrorMessage={swapErrorMessage}
-          swappableAssets={swappableAssets.filter(isNotNull)}
-          wavesFeeCoins={wavesFeeCoins}
-          onSwap={onPrepare}
-        />
-      )}
-    </SignWrapper>
+      }
+    },
+    [assets, selectedAccount, t, wavesFeeCoins]
+  );
+
+  const { sign, pending: signPending } = useSign(onConfirm);
```

Comment:
May be name it `signPending` (or better `isSignPending`) in `useSign` instead of renaming on the call site? It was the whole point of changing from array to object as a return value, to have better name already chosen for us :-)